File: objc/NSOperationObjCImageLoader/NSOperationObjCImageLoader/PhotoRecord.m
//
//  PhotoRecord.m
//  NSOperationObjCImageLoader
//
//  Created by zhen gong on 8/27/17.
//  Copyright © 2017 zhen gong. All rights reserved.
//

#import "PhotoRecord.h"

@implementation PhotoRecord

- (instancetype)initWith:(NSString *)name url:(NSURL *)url {
    if(self = [super init]) {
        _name = name;
        _url = url;
        _image = [UIImage imageNamed:@"dribbble_ball"];
    }
    return self;
}

@end

File: swift/NSOperationSwift/NSOperationSwift/Models/PhotoRecord.swift

//
//  PhotoRecord.swift
//  NSOperationSwift
//
//  Created by zhen gong on 6/30/17.
//  Copyright © 2017 zhen gong. All rights reserved.
//

import Foundation
import UIKit

class PhotoRecord {
    let name: String
    let url:URL
    var state = PhotoRecordState.New
    var image = UIImage(named: "Placeholder")
    
    init(name:String, url:URL) {
        self.name = name
        self.url = url
    }
}
